### PR TITLE
feat: manipulate common component (MN-262)

### PR DESCRIPTION
**Pull request by deepakglobant**
Manipulate common component

### CODE REVIEW

File: packages/inspection-report/src/components/common/DamageManipulator.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  },
+});
+
+function DamageManipulator({ damageMode, displayMode, onConfirm, damage }) {
+  const { t } = useTranslation();
+
+  const [isEnabled, setIsEnabled] = useState(false);
+  const toggleSwitch = () => setIsEnabled((previousState) => !previousState);
+  const [severity, setSeverity] = useState('');
+  const [pricing, setPricing] = useState(0);
```
Comment (souyahia-monk) [inline packages/inspection-report/src/components/common/DamageManipulator.js:81]: don't declare these states. These values are located in the damage prop

File: packages/inspection-report/src/components/common/DamageManipulator.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    borderTopRightRadius: 12,
+    height: 16,
+    width: 8,
+  },
+});
+
+function DamageManipulator({ damageMode, displayMode, onConfirm, damage }) {
+  const { t } = useTranslation();
+
+  const [isEnabled, setIsEnabled] = useState(false);
```
Comment (souyahia-monk) [inline packages/inspection-report/src/components/common/DamageManipulator.js:78]: if you take a look at the specs on Jira you should have something like :

```javascript
const [isEnabled, setIsEnabled] = useState(!!damage);
```

File: packages/inspection-report/src/components/common/DamageManipulator.js (hunk 1 above)
Comment (souyahia-monk) [inline packages/inspection-report/src/components/common/DamageManipulator.js:81]: No this is not how you are supposed to implement this. If you take a look at the docs you should have an edited damage as a state and use the `damage` prop as the initial value. Please do something like :

```javascript
// Declare this OUTSIDE of the component
const initialDamage = {
  severity: { severity: 'low' },
  pricing: { pricing: 0 },
  all: { severity: 'low', pricing: 0 },
};

// Inside the component
const [editedDamage, setEditedDamage] = useState(damage);
const initialDamageValue = useMemo(() => {
  if (damageMode === 'severity') {
    return 
}, []);
const toggleSwitch = useCallback(
  () => setEditedDamage((dmg) => (dmg ? null : initialDamage[damageMode])),
  [damageMode],
);
```

And then the value of the slider should be linked to `editedDamage.pricing` and the selected severity should be linked to `editedDamage.severity`. Also, to know wether or not the checkbox is checked or not, you just have to check if `editedDamage` is defined

File: packages/inspection-report/src/components/common/DamageManipulator.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  const [pricing, setPricing] = useState(0);
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.damageWrapper}>
+        <View>
+          <Text style={styles.title}>{t('damageManipulator.damages')}</Text>
+          <Text style={{ fontSize: 16, lineHeight: 24 }}>{t('damageManipulator.notDamaged')}</Text>
+        </View>
+        <SwitchButton onPress={toggleSwitch} isEnabled={isEnabled} />
```
Comment (souyahia-monk) [inline packages/inspection-report/src/components/common/DamageManipulator.js:90]: as I said in the commment above, you should have `isEnabled={!!editedDamage}`

File: packages/inspection-report/src/components/common/DamageManipulator.js (hunk 1 above)
Comment (deepakglobant) [reply packages/inspection-report/src/components/common/DamageManipulator.js:81]: ah okay. Thanks

Comment (deepakglobant) [reply packages/inspection-report/src/components/common/DamageManipulator.js]: Resolved

File: packages/inspection-report/src/components/common/DamageManipulator.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            <View style={{ flexDirection: 'row', alignItems: 'center', marginTop: 8 }}>
+              <Slider
+                style={{ marginRight: 15 }}
+                minimumValue={0}
+                maximumValue={9999}
+                lowerLimit={0}
+                upperLimit={9999}
+                step={1}
+                disabled={displayMode === 'full' && !editedDamage}
+                value={editedDamage?.pricing || 0}
```
Comment (souyahia-monk) [inline packages/inspection-report/src/components/common/DamageManipulator.js:155]: use `??` operator instead

File: packages/inspection-report/src/components/common/DamageManipulator.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+                thumbTintColor="#8da8ff"
+                minimumTrackTintColor="#5d5e67"
+                maximumTrackTintColor="#000000"
+                onValueChange={(value) => setEditedDamage((dmg) => ({ ...dmg, pricing: value }))}
+              />
+              <Text>
+                {editedDamage?.pricing || 0}
```
Comment (souyahia-monk) [inline packages/inspection-report/src/components/common/DamageManipulator.js:162]: use `??` operator instead

Comment (deepakglobant) [reply packages/inspection-report/src/components/common/DamageManipulator.js]: Done

**Review by souyahia-monk — APPROVED** [review]

**Pull request merged**